plugins/qc-s5gen2/fu-qc-s5gen2-hid-device.c

```diff
@@ -210,6 +210,51 @@ fu_qc_s5gen2_hid_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_qc_s5gen2_hid_device_setup(FuDevice *device, GError **error)
+{
+	guint idx;
+	guint16 vid = fu_device_get_vid(device);
+	guint16 pid = fu_device_get_pid(device);
+
+	/* FuHidDevice->setup */
+	if (!FU_DEVICE_CLASS(fu_qc_s5gen2_hid_device_parent_class)->setup(device, error))
+		return FALSE;
+
+	fu_device_add_instance_u16(device, "VID", vid);
+	fu_device_add_instance_u16(device, "PID", pid);
+
+	idx = fu_usb_device_get_manufacturer_index(FU_USB_DEVICE(device));
+	if (idx != 0x00) {
+		g_autofree gchar *tmp = NULL;
+		tmp = fu_usb_device_get_string_descriptor(FU_USB_DEVICE(device), idx, NULL);
+		fu_device_add_instance_str(device, "MANUFACTURER", tmp);
+		g_debug("Manufacturer: %s", tmp);
```

Comment:
I don't think this is helpful, it's in the instance ID already.

Reply:
Very helpful. With `no-generic-guids` flag only GUID shown without the instance ID.
i.e.:
```
└─CCC5171X:
      Device ID:          fbd12441fe94a4d805ff32edce5dc1580a71aff2
      Current version:    1.0.1
      Vendor:             Cambridge Silicon Radio, Ltd (USB:0x0A12)
      Serial Number:      5B23541510FF00B52000
      GUID:               9fe24e67-3d4c-5ed5-8f13-239b6891db9e
      Device Flags:       • Updatable
                          • Device stages updates
                          • Device can recover flash failures
                          • Device is usable for the duration of the update
                          • Signed Payload
                          • Can tag for emulation
```

Reply:
I think that's probably `g_ptr_array_set_size(fwupd_device_get_instance_ids(FWUPD_DEVICE(self)), 0);` being overly simplistic. Will ponder.